src/audio/volume/volume.c

```diff
@@ -408,57 +422,53 @@ static int volume_ctrl_set_cmd(struct comp_dev *dev,
 				      "cdata->comp_id = %u",
 				      cdata->comp_id);
 		for (j = 0; j < cdata->num_elems; j++) {
-			trace_volume_with_ids(dev, "volume_ctrl_set_cmd(), "
-					      "SOF_CTRL_CMD_VOLUME, "
-					      "channel = %u, value = %u",
-					      cdata->chanv[j].channel,
-					      cdata->chanv[j].value);
-			i = cdata->chanv[j].channel;
-			if (i >= 0 && i < SOF_IPC_MAX_CHANNELS) {
-				ret = volume_set_chan(dev, i,
-						      cdata->chanv[j].value);
+			ch = cdata->chanv[j].channel;
+			val = cdata->chanv[j].value;
+			trace_volume_with_ids(dev, "volume_ctrl_set_cmd(), channel = %u"
+					      ", value = %u", ch, val);
+			if (ch >= 0 && ch < SOF_IPC_MAX_CHANNELS) {
+				if (cd->muted[ch])
+					cd->mvolume[ch] = val;
+				else
+					ret = volume_set_chan(dev, ch, val);
 			} else {
 				trace_volume_error_with_ids(dev,
-							    "volume_ctrl_set_cmd() "
-							    "error: "
-							    "SOF_CTRL_CMD_VOLUME, "
-							    "invalid i = %u",
-							    i);
+							    "volume_ctrl_set_cmd(), illegal channel = %u",
```

Comment:
`ch` is defined as `int` and you check it for `< 0`, so, I think `%d` is a better choice here.

Reply:
Yep

Reply:
done